x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -75,7 +76,11 @@ export interface FetchResult {
 
 export type BulkUpdateResult = Result<
   ConcreteTaskInstance,
-  { entity: ConcreteTaskInstance; error: Error }
+  { type: string; id: string; error: SavedObjectError }
```

Comment:
Doing two things here:
- Removing `entity` when an update error occurs on a document.
- Fixing `error: Error` because it was really `type: string; id: string; error: SavedObjectError` under the hood. I needed this to access `.error.statusCode` properly.

These changes allows to keep the `retryableBulkUpdate` response the same shape as the regular BulkUpdateResult in scenarios `bulkGet` cannot find a document.